pkg/validation/validation_test.go

```diff
@@ -11,14 +11,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package validation
+package validation_test
```

Comment:
what was the thinking behind moving these tests to a separate package for this PR? I'm not necessarily opposed but I don't think we're doing this in other places and my top preference is usually to be consistent across the code base.

Reply:
We need a single place for this logic to live since now two controllers are going to be doing the same logic on the same resources and those checks need to be identical. 

If for instance one controller says that the validation is fine, but another doesn't then they conflict and might not get written out to the user in a status. 

I'm not sure if that's moving all the code from `projectcontour/contour/internal` to `projectcontour/contour/pkg` or a separate project (i.e. `projectcontour/api`).

Note that the reasons we've kept that logic in `internal` was to allow us free will to change up the interfaces and logic (since internal isn't importable), but as soon as we publish those out we end up with more deprecation & integration work.